pkgs/development/libraries/gstreamer/good/extra-modules.sh

```diff
@@ -0,0 +1,8 @@
+addGIOModulesLibPath () {
+    if [[ -d "$1/lib/gio/modules" && "$1" == *glib-networking* ]]
+    then
+        export GIO_EXTRA_MODULES="${GIO_EXTRA_MODULES-}${GIO_EXTRA_MODULES:+:}$1/lib/gio/modules"
```

Comment:
```suggestion
        addToSearchPath GIO_EXTRA_MODULES "$1/lib/gio/modules"
```

There's a helper function in stdenv to do this

I do almost wonder if this setup hook should be on `glib` or `glib-networking` too 🤔

Reply:
thanks. `addToSearchPath` is perfect.

[glib/setup-hook.sh](https://github.com/NixOS/nixpkgs/blob/79fee6027b432958825b0bb83a1538314327a8a6/pkgs/development/libraries/glib/setup-hook.sh) does seem like a better fit.

Reply:
or maybe `glib-networking` is better for GIO_EXTRA_MODULES setup-hook. I'm less excited about polluting the env space with GIO_EXTRA_MODULES for all packages which use glib.

Reply:
`glib-networking` is just one package providing a gio module so it does not feel right. We could make a separate setup hook and propagate it from `glib-networking` if we want.